starport/pkg/localfs/search_test.go

```diff
@@ -0,0 +1,115 @@
+package localfs
+
+import (
+	"io/fs"
+	"os"
+	"path/filepath"
+	"syscall"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func setupGlobTests(t *testing.T, files []string) string {
+	t.Helper()
+	dir, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+	tmpdir, err := os.MkdirTemp(dir, "glob-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() {
+		_ = os.RemoveAll(tmpdir)
+	})
+	for _, file := range files {
+		fileDir := filepath.Dir(file)
+		fileDir = filepath.Join(tmpdir, fileDir)
+		if _, err := os.Stat(fileDir); os.IsNotExist(err) {
+			err = os.MkdirAll(fileDir, 0755)
+		}
+		err = os.WriteFile(filepath.Join(tmpdir, file), []byte{}, 0644)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	return tmpdir
+}
+
+func TestSearch(t *testing.T) {
+	files := []string{
+		"foo/file.proto",
+		"foo/bar/file1.proto",
+		"foo/bar/file2.proto",
+		"foo/baz/file.proto",
+		"foo/file",
+		"foo/baz/file",
+	}
+	tmpdir := setupGlobTests(t, files)
+	type args struct {
+		path    string
+		pattern string
+	}
+	tests := []struct {
+		name string
+		args args
+		want []string
+		err  error
+	}{
+		{
+			name: "get all proto files by pattern",
+			args: args{
+				path:    tmpdir,
+				pattern: "*.proto",
+			},
+			want: []string{
+				tmpdir + "/foo/bar/file1.proto",
```

Comment:
Better if we can use `filepath.Join` here or we can add this tmpdir in the for loop below. This way we don't need to add it for all cases individually.

Reply:
I add the `filepath.Join` in the table test cases because the iteration below it's for the test cases. We need to create a new for only to make the joins into the expected cases.